StrataTest/Languages/Boogie/DatatypeVerificationTests.lean

```diff
@@ -0,0 +1,613 @@
+/-
+  Copyright Strata Contributors
+
+  SPDX-License-Identifier: Apache-2.0 OR MIT
+-/
+
+import Strata.Languages.Boogie.Boogie
+import Strata.Languages.Boogie.Verifier
+import Strata.DL.Lambda.TypeFactory
+
+/-!
+# Datatype Verification Integration Tests
+
+Verify Boogie programs with datatypes, encoding with declare-datatype
+-/
+
+namespace Boogie.DatatypeVerificationTests
+
+open Lambda
+open Std (ToFormat Format)
+open Imperative
+
+/-! ## Test Datatypes -/
+
+/-- Simple Option datatype: Option a = None | Some a -/
+def optionDatatype : LDatatype Visibility :=
+  { name := "Option"
+    typeArgs := ["a"]
+    constrs := [
+      { name := ⟨"None", .unres⟩, args := [], testerName := "isNone" },
+      { name := ⟨"Some", .unres⟩, args := [(⟨"OptionVal", .unres⟩, .ftvar "a")], testerName := "isSome" }
+    ]
+    constrs_ne := by decide }
+
+/-- Recursive List datatype: List a = Nil | Cons a (List a) -/
```

Comment:
I would like to see in this comment the names of the destructors `hd` and `tl`, as they are relevant later.